viscy/data/triplet.py

```diff
@@ -572,4 +566,46 @@ def predict_dataloader(self):
             persistent_workers=self.persistent_workers,
             drop_last=False,
             pin_memory=self.pin_memory,
+            collate_fn=lambda x: x,
+        )
+
+    def _final_crop(self) -> BatchedCenterSpatialCropd:
+        """Setup final cropping: center crop to the target size."""
+        return BatchedCenterSpatialCropd(
+            keys=self.source_channel,
+            roi_size=(
+                self.z_window_size,
+                self.yx_patch_size[0],
+                self.yx_patch_size[1],
+            ),
         )
+
+    def _find_transform(self, key: str):
+        if self.trainer:
+            if self.trainer.predicting:
+                return self._no_augmentation_transform
+        # NOTE: for backwards compatibility
+        if key == "anchor" and self.time_interval in ("any", 0):
+            return self._no_augmentation_transform
+        return self._augmentation_transform
+
+    def on_after_batch_transfer(self, batch, dataloader_idx: int):
+        """Apply transforms after transferring to device."""
+        if isinstance(batch, Tensor):
+            # example array
+            return batch
+        for key in ["anchor", "positive", "negative"]:
+            if key in batch:
+                norm_meta_key = f"{key}_norm_meta"
```

Comment:
here we are enforcing that all the dataset always run pre-processing. Should that be always true?

Reply:
For our current use case I think that's always true. We can change this if a need arises.